src/SlackMessage.js

```diff
@@ -21,37 +22,35 @@ class SlackMessageRoot {
 }
 
 class SlackMessage extends SlackMessageRoot {
-  #_token = "";
   #_channel = "";
   #_ts = "";
+  #_slackGateway;
 
-  constructor(token, message = { channel: "", ts: "" }) {
+  constructor(slackToken, message = { channel: "", ts: "" }) {
     super();
-    if (!token) throw new Error("token is required");
-    this.#_token = token;
+    this.#_slackGateway = new SlackGateway(slackToken);
```

Comment:
The SlackMessage class will delegate interaction with the slack api to the SlackGateway